From 90748a2e0b411ee48249897e36389ca51272554a Mon Sep 17 00:00:00 2001
From: Yuan Juntao <juntao.yuan@intel.com>
Date: Thu, 2 Apr 2015 18:31:33 +0800
Subject: [PATCH] mmc: reduce mmc resume latency.

Reduce two delay duration in mmc driver in order to reduce mmc resume
latency.
This patch can improve suspend/resume performance.

Change-Id: Id9c514f83ae75b1569b2ee302a992f9b22c9e0cd
Signed-off-by: Yuan Juntao <juntao.yuan@intel.com>
---
 drivers/mmc/core/core.c    | 2 +-
 drivers/mmc/core/mmc_ops.c | 8 ++++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index 09f0b49aab9c..753989171b1b 100644
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -1691,7 +1691,7 @@ void mmc_power_up(struct mmc_host *host, u32 ocr)
 	 * This delay must be at least 74 clock sizes, or 1 ms, or the
 	 * time required to reach a stable voltage.
 	 */
-	mmc_delay(10);
+	mmc_delay(1);
 
 	mmc_host_clk_release(host);
 }
diff --git a/drivers/mmc/core/mmc_ops.c b/drivers/mmc/core/mmc_ops.c
index e5b5eeb548d1..b11977e6429e 100644
--- a/drivers/mmc/core/mmc_ops.c
+++ b/drivers/mmc/core/mmc_ops.c
@@ -141,7 +141,7 @@ int mmc_send_op_cond(struct mmc_host *host, u32 ocr, u32 *rocr)
 	cmd.arg = mmc_host_is_spi(host) ? 0 : ocr;
 	cmd.flags = MMC_RSP_SPI_R1 | MMC_RSP_R3 | MMC_CMD_BCR;
 
-	for (i = 100; i; i--) {
+	for (i = 1000; i; i--) {
 		err = mmc_wait_for_cmd(host, &cmd, 0);
 		if (err)
 			break;
@@ -161,7 +161,11 @@ int mmc_send_op_cond(struct mmc_host *host, u32 ocr, u32 *rocr)
 
 		err = -ETIMEDOUT;
 
-		mmc_delay(10);
+		/* Acorrding to Spec, it is OK to poll busy
+		 * bit by ACMD41 less than 50ms intervals.
+		 * Move interval to 1ms for quick response.
+		 */
+		mmc_delay(1);
 	}
 
 	if (rocr && !mmc_host_is_spi(host))
-- 
2.1.0

